packages/core/src/components/TextArea/__tests__/TextArea.test.tsx

```diff
@@ -99,4 +100,57 @@ describe("TextArea", () => {
       expect(textarea).toHaveAttribute("aria-describedby", "help-text-textarea-help-text");
     });
   });
+
+  describe("TextArea character count and limit", () => {
+    it("should not show limit when character limit is not provided", () => {
+      render(<TextArea />);
+      expect(screen.queryByText("0/")).not.toBeInTheDocument();
+    });
+
+    it("should prevent typing when character limit is reached", () => {
+      const handleChange = jest.fn();
+      render(<TextArea maxLength={5} allowExceedingLimit={false} onChange={handleChange} />);
+
+      const charCount = screen.getByText("0/5");
+      expect(charCount).toBeInTheDocument();
+      
+
+      const input = screen.getByRole("textbox");
+      userEvent.type(input, "12345a");
+
+      expect(input).toHaveValue("12345");
+      expect(charCount).toHaveTextContent("5/5");
+      expect(handleChange).toHaveBeenCalledTimes(5);
+    });
+
+    it("should allow typing when limit is allowed to be exceeded", () => {
+      render(<TextArea maxLength={10} allowExceedingLimit={true} />);
+
+      const charCount = screen.getByText("0/10");
+      expect(charCount).toBeInTheDocument();
+
+      const input = screen.getByRole("textbox");
+      userEvent.type(input, "12345678910");
+
+      expect(input).toHaveValue("12345678910");
+      expect(charCount).toHaveTextContent("11/10");
+    });
+
+    it("should allow text removal when character limit is exceeded", async () => {
+      render(<TextArea maxLength={5} allowExceedingLimit={false} />);
+
+      const input = screen.getByRole("textbox");
+      userEvent.type(input, "12345");
+
+      // Attempt to add more characters (should be prevented)
+      userEvent.type(input, "a");
+      expect(input).toHaveValue("12345");
+
+      // Remove a character (should be allowed)
+      userEvent.type(input, "{backspace}");
+      userEvent.type(input, "4");
+
+      expect(input).toHaveValue("12344");
+    });
```

Comment:
Let's please add a test that validates that we can't input more characters if we met the limit

Reply:
https://github.com/mondaycom/vibe/pull/2574/files#diff-ab35f45e7f2a0be5ce7d34564c0a408202d6e74da27f72651a6863fce4e4a29cR110-R124

This test validates that!